src/main/java/ch/naviqore/service/PublicTransportService.java

```diff
@@ -0,0 +1,31 @@
+package ch.naviqore.service;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+
+public interface PublicTransportService {
+    @NotNull List<StopTime> getNextDepartures(@NotNull Stop stop, @NotNull LocalDate from, @Nullable LocalDate until, int limit);
+
+    @NotNull List<Stop> getStops(@NotNull String like, @NotNull SearchType searchType);
+
+    @Nullable Stop getClosestStop(@NotNull Location location);
```

Comment:
Keep the interface as lean as possible, I would not force the service implementation to offer methods with default values, since then it is not anymore on the layer of the service to control the default values and every implementation will set their own default. Just enforce the fully specified method with all parameters, we can then set default in the REST API and they will stay the same, independent of the service implementation we use.

Reply:
Where do you see a default value?